terraform/applications/discourse/iam.tf

```diff
@@ -1,39 +1,14 @@
-resource "aws_iam_role" "discourse_role" {
-  name               = "discourse-${var.environment}"
-  path               = "/discourse/"
-  assume_role_policy = data.aws_iam_policy_document.allow_assume_role.json
+locals {
+  cluster_workers_role_name = replace(
+    data.terraform_remote_state.k8s.outputs.cluster_worker_iam_role_arn,
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/",
+    ""
+  )
```

Comment:
Our EKS module [already exports the role name](https://github.com/mozilla-it/terraform-modules/blob/33c9f55b23035e130291e13f0f4bab542736727b/aws/eks/outputs.tf#L37-L39), so if we re-export the name [in the shared Terraform code in this repo](https://github.com/mozilla-it/itse-apps-prod-1-infra/blob/083c744e7760f1a370ef776adfeeac78700af50a/terraform/shared/outputs.tf) we can make do without this hack.

Reply:
@smarnach feel free to edit the commit here and the one in it’s-apps-stage-infra-1 to show your requested changes and ensure the role names are indeed the same (we often need a workers role specifically).

Reply:
You are of course right that the role name I linked was the cluster IAM role, not the worker IAM role. I had to file https://github.com/mozilla-it/terraform-modules/pull/160 to add the worker IAM role as an additional output.

Reply:
thanks @smarnach ! will follow up on this PR now, thanks for getting it all aligned for me